test/org/zaproxy/zap/extension/pscanrulesAlpha/HashDisclosureScannerUnitTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Plugin.AlertThreshold;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * Unit test for {@link HashDisclosureScanner}.
+ */
+public class HashDisclosureScannerUnitTest extends PassiveScannerTest<HashDisclosureScanner> {
+
+    @Override
+    protected HashDisclosureScanner createScanner() {
+        return new HashDisclosureScanner();
+    }
+
+    @Before
+    public void before() throws HttpMalformedHeaderException {
+        rule.setAlertThreshold(AlertThreshold.LOW); // Required by MD4/MD5 tests
+    }
+
+    @Test
+    public void shouldRaiseAlertWhenResponseContainsUpperHash() throws Exception {
```

Comment:
The names should be explicit about the hash being tested (it has many).

Reply:
Yup that makes sense. Will fix up.